projects/fluid_simulation/sim.cpp

```diff
@@ -28,32 +28,32 @@ const ppx::grfx::Format kR    = ppx::grfx::FORMAT_R16_FLOAT;
 const ppx::grfx::Format kRG   = ppx::grfx::FORMAT_R16G16_FLOAT;
 const ppx::grfx::Format kRGBA = ppx::grfx::FORMAT_R16G16B16A16_FLOAT;
 
-const SimulationConfig& FluidSimulation::GetConfig() const
+ppx::Result FluidSimulation::Create(ppx::Application* app, ppx::grfx::DevicePtr device, ppx::uint2 resolution, const SimulationConfig& config, std::unique_ptr<FluidSimulation>* pSim)
 {
-    return mApp->GetSimulationConfig();
+    *pSim = std::make_unique<FluidSimulation>(app, device, resolution, config);
+    return (*pSim)->Initialize();
 }
 
-FluidSimulation::FluidSimulation(ProjApp* app)
-    : mApp(app)
+ppx::Result FluidSimulation::Initialize()
 {
     // Create descriptor pool shared by all pipelines.
     ppx::grfx::DescriptorPoolCreateInfo dpci = {};
     dpci.sampler                             = 1024;
     dpci.sampledImage                        = 1024;
     dpci.uniformBuffer                       = 1024;
     dpci.storageImage                        = 1024;
-    PPX_CHECKED_CALL(GetApp()->GetDevice()->CreateDescriptorPool(&dpci, &mDescriptorPool));
+    PPX_CHECKED_CALL(GetDevice()->CreateDescriptorPool(&dpci, &mDescriptorPool));
```

Comment:
This seems like it can be leaner since we know the number of simulation frames and the number of resources we'll use.

Reply:
Yup. You want this change in this PR or a subsequent one? This is the first of several cleanups.